### DISCUSSION TITLE
Missing/cut logs?

### DISCUSSION THREAD

**Discussion by jasonkuhrt** [vercel/vercel#6502] [Help] [answered]
We've noticed some issues with logging where we think logs do not show up in the log view of Vercel. We've also noticed that log splitting is an issue about Vercel that Log Flare says it "fixes" with their logger integration and node library. My questions are as follows:

- Have you seen lost/cut logs
- Are these AWS-based issues or Vercel-based issues?
- Does adding a log drain integration to Vercel solve the log loss and/or log cuts?
- Is there documentation anywhere online about this?

**Comment by leerob** [comment, marked as answer]
Do you have an example of lost logs? As a quick reminder, Vercel does not persist logs (and this wouldn't be related to AWS).

We do have documentation on this! As well as how to use Log Drains.

https://vercel.com/docs/logs

**Comment by jasonkuhrt** [comment]
Part of the answer here is the 4kb limit.

**Reply by statico** [reply]
I'm having this issue and found this thread. Is the logging limit documented anywhere? I can't seem to find it.

We're sending our logs to logtail.com where we only appear to be getting the last ~4 KB of each function's logs.
